src/FileRouter.php

```diff
@@ -155,61 +155,57 @@ private function parseRequestPath(ServerRequestInterface $request): Generator
             }
         }
 
-        $possibleAction = null;
-
         if ($path === '/') {
             yield [
-                $this->cleanClassname(
-                    $this->namespace . '\\' . $this->baseControllerDirectory . '\\' . $this->defaultControllerName . $this->classPostfix
-                ),
-                $possibleAction,
+                $this->makeClassName($this->defaultControllerName),
+                null,
             ];
             return;
         }
 
         $controllerName = preg_replace_callback(
-            '#(/.)#u',
-            static fn(array $matches) => mb_strtoupper($matches[1]),
+            '#/.#u',
+            static fn(array $matches) => mb_strtoupper($matches[0]),
             $path,
         );
 
-        if (!preg_match('#^(.*?)/([^/]+)/?$#', $controllerName, $matches)) {
+        if (!preg_match('#^/?(.*?)/([^/]+)/?$#', $controllerName, $matches)) {
```

Comment:
Why?

Reply:
Here `$controllerName` contains path with uppercased first symbols. Renamed it to `$pathAsNamespace`.